src/screens/EventsScreen/FilterHeader.js

```diff
@@ -20,31 +20,54 @@ export type Props = {
   onFilterButtonPress: () => void,
   onDateFilterButtonPress: () => void,
   selectedCategories: Set<EventCategoryName>,
-  numTagFiltersSelected: number
+  numTagFiltersSelected: number,
+  resetAllFiltersPress: () => void
 };
 
 class FilterHeader extends React.PureComponent<Props> {
+  static defaultProps = {
+    resetAllFiltersPress: () => {},
+    selectedCategories: new Set()
+  };
+
   render() {
     const {
       dateFilter,
       onFilterCategoriesPress,
       selectedCategories,
       onFilterButtonPress,
       onDateFilterButtonPress,
-      numTagFiltersSelected
+      numTagFiltersSelected,
+      resetAllFiltersPress
     } = this.props;
     const formattedDateFilter = dateFilter
       ? formatDateRange(dateFilter)
       : text.selectDates;
 
+    const anyAppliedFilters: boolean =
+      !!dateFilter || numTagFiltersSelected > 0 || selectedCategories.size > 0;
+
     return (
       <View accessibilityTraits={["header"]} style={styles.container}>
         <ContentPadding>
-          <View testID="event-filter-header" style={styles.content}>
-            <FilterHeaderCategories
-              onFilterPress={onFilterCategoriesPress}
-              selectedCategories={selectedCategories}
-            />
+          <View>
+            {anyAppliedFilters && (
+              <View style={styles.clearAllWrapper}>
+                <FilterHeaderButton
+                  active={false}
+                  text="Reset all filters"
+                  label="Reset all filters"
+                  style={styles.clearAll}
+                  onPress={resetAllFiltersPress}
+                />
+              </View>
+            )}
+            <View testID="event-filter-header" style={styles.content}>
+              <FilterHeaderCategories
+                onFilterPress={onFilterCategoriesPress}
+                selectedCategories={selectedCategories}
+              />
+            </View>
```

Comment:
I think I might pull this out into a separate component. Enough logic in here to make this a separate thing.